### ISSUE TITLE
Reproducing MoCo-v2 Results with CARE

### ISSUE THREAD

**Issue by KimManjin** [Sharut/CARE#1]
Hi, congratulations for the great work!
Firstly, I would like to extend my congratulations on your remarkable work. It's truly inspiring.

I've been currently working on reproducing the results of the paper, e.g., SimCLR and MoCo-v2 on ImageNet100.
I successfully did reproduce the results of SimCLR, but failed reproducing those of MoCo-v2.

I run the provided training scripts and upload the training curves in the below.

Following the training scripts provided, I observed a performance degradation when applying CARE to MoCo-v2, contrary to the improvements seen with SimCLR. 
Notably, there was a decrease in KNN accuracy and linear probing accuracy for MoCo-v2 (from 73.8 to 73.3). I have attached the training curves below for your reference:

![image](https://github.com/Sharut/CARE/assets/40816028/4ceccd24-2639-460d-9407-15f0a23694ff)

Could you check if the training scripts of MoCo v2 is correct?
Additionally, if the scripts are correct, I would greatly appreciate any insights or hypotheses you might have regarding the observed discrepancies.

Thank you for your time and consideration. I look forward to your guidance and insights.

Thanks.
